src/App.js

```diff
@@ -1,43 +1,87 @@
 import React from "react";
 import Joke from "./components/Joke";
 import PageHeader from "./components/PageHeader";
+import SearchInput from "./components/SearchInput";
+import PropTypes from "prop-types";
 
 class App extends React.Component {
-  constructor() {
-    super()
-    this.state = {
-      jokeData: {
-        results: []
-      }
+  state = {
+    jokeData: {
+      results: []
+    },
+    searchTerm: "",
+    loadingState: "resting"
```

Comment:
It would be a good idea to turn the possible values of `loadingState` into an enumeration so you don't need to make sure you type and spell it correctly everywhere you use it.

Reply:
So would I add a PropType here such as
`loadingState: PropTypes.oneOf(['resting', 'loaded', 'loading']).isRequired`

and then set a variable such as
`const message = {resting: "resting", loaded: "loaded"}` etc

So that way I could use `state = {loadingState: {loaded: message.loaded, resting: message.resting}` etc?

Reply:
Kind of... since App.js is the main entry point into your application, nothing really passes any props into it, so don't worry about PropTypes.

An enum (enumeration) would look something like this:
```
const LOADING_STATE = { RESTING: 'resting', LOADED: 'loaded', loading: 'loading', ...}
```
and then you'd use it like `LOADING_STATE.LOADED` etc.